src/overlays/actors/ovl_Obj_Kibako2/z_obj_kibako2.c

```diff
@@ -31,21 +33,143 @@ const ActorInit Obj_Kibako2_InitVars = {
     (ActorFunc)ObjKibako2_Update,
     (ActorFunc)ObjKibako2_Draw,
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Obj_Kibako2/func_80B959D0.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Obj_Kibako2/func_80B95A28.s")
+static ColliderCylinderInit sCylinderInit = {
+    { COLTYPE_UNK10, 0x00, 0x09, 0x00, 0x20, COLSHAPE_CYLINDER },
+    { 0x00, { 0x00000000, 0x00, 0x00 }, { 0x40000040, 0x00, 0x00 }, 0x00, 0x01, 0x00 },
+    { 31, 48, 0, { 0, 0, 0 } },
+};
+
+static InitChainEntry sInitChain[] = {
+    ICHAIN_VEC3F_DIV1000(scale, 100, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneForward, 3000, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneScale, 500, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneDownward, 1000, ICHAIN_STOP),
+};
+
+void ObjKibako2_InitCollider(Actor* thisx, GlobalContext* globalCtx) {
+    ObjKibako2* this = THIS;
+
+    Collider_InitCylinder(globalCtx, &this->collider);
+    Collider_SetCylinder(globalCtx, &this->collider, &this->dyna.actor, &sCylinderInit);
+    Collider_CylinderUpdate(&this->dyna.actor, &this->collider);
+}
+
+void ObjKibako2_Break(ObjKibako2* this, GlobalContext* globalCtx) {
+    s32 padE4;
+    s32 padE0;
+    PosRot* thisPosRot;
+    Vec3f pos;
+    Vec3f velocity;
+    s16 angle;
+    s32 i;
+
+    thisPosRot = &this->dyna.actor.posRot;
+    for (i = 0, angle = 0; i < 0x10; i++, angle += 0x4E20) {
+        f32 sn = Math_SinS(angle);
+        f32 cs = Math_CosS(angle);
+        f32 temp_rand;
+        s32 phi_s0;
+
+        temp_rand = Rand_ZeroOne() * 30.0f;
+        pos.x = sn * temp_rand;
+        pos.y = (Rand_ZeroOne() * 10.0f) + 2.0f;
+        pos.z = cs * temp_rand;
+        velocity.x = pos.x * 0.2f;
+        velocity.y = (Rand_ZeroOne() * 10.0f) + 2.0f;
+        velocity.z = pos.z * 0.2f;
+        pos.x += thisPosRot->pos.x;
+        pos.y += thisPosRot->pos.y;
+        pos.z += thisPosRot->pos.z;
+        temp_rand = Rand_ZeroOne();
+        if (temp_rand < 0.05f) {
+            phi_s0 = 0x60;
+        } else if (temp_rand < 0.7f) {
+            phi_s0 = 0x40;
+        } else {
+            phi_s0 = 0x20;
+        }
+        EffectSsKakera_Spawn(globalCtx, &pos, &velocity, &pos, -200, phi_s0, 28, 2, 0, (Rand_ZeroOne() * 30.0f) + 5.0f,
+                             0, 0, 70, KAKERA_COLOR_NONE, OBJECT_KIBAKO2, D_06001000);
+    }
+    func_80033480(globalCtx, &thisPosRot->pos, 90.0f, 6, 100, 160, 1);
+}
+
+void ObjKibako2_SpawnCollectible(ObjKibako2* this, GlobalContext* globalCtx) {
+    s16 itemDropped;
+    s16 collectibleFlagTemp;
+
+    collectibleFlagTemp = this->collectibleFlag;
+    itemDropped = this->dyna.actor.initPosRot.rot.x;
+    if (itemDropped >= 0 && itemDropped < 0x1A) {
+        Item_DropCollectible(globalCtx, &this->dyna.actor.posRot.pos, itemDropped | (collectibleFlagTemp << 8));
+    }
+}
+
+void ObjKibako2_Init(Actor* thisx, GlobalContext* globalCtx) {
+    ObjKibako2* this = THIS;
+    s16 pad;
+    s32 sp2C;
+    u32 bgId;
+
+    sp2C = 0;
```

Comment:
with the new bgcheck merge you can type this as a CollisionHeader* and set it to null. can even do it on the same line as the declaration if you like